src/main/java/online/talkandtravel/service/impl/EventServiceImpl.java

```diff
@@ -136,28 +148,32 @@ private void removeConnections(EventRequest request, Chat chat, User author) {
       throw new UserNotJoinedTheChatException(request.authorId(), request.chatId());
     }
 
-    UserCountry userCountry =
-        userCountryRepository
-            .findByCountryNameAndUserId(chat.getCountry().getName(), author.getId())
-            .orElseThrow(
-                () ->
-                    new UserCountryNotFoundException(chat.getCountry().getName(), author.getId()));
-
     // remove record from userChats
     userChatRepository.delete(optionalUserChat.get());
 
-    // check if there is no records in UserChats with userId and CountryName,
-    // if true, remove record from userCountries
-    List<UserChat> userChats =
-        userChatRepository.findAllByUserIdAndUserCountryId(request.authorId(), userCountry.getId());
-    if (userChats.isEmpty()) {
-      // if this was the last chat in country for user, we remove connection with Country
-      userCountryRepository.delete(userCountry);
+    if (chat.getCountry() != null) {
+      UserCountry userCountry =
+          userCountryRepository
+              .findByCountryNameAndUserId(chat.getCountry().getName(), author.getId())
+              .orElseThrow(
+                  () ->
+                      new UserCountryNotFoundException(
+                          chat.getCountry().getName(), author.getId()));
+
+      // check if there is no records in UserChats with userId and CountryName,
+      // if true, remove record from userCountries
+      List<UserChat> userChats =
+          userChatRepository.findAllByUserIdAndUserCountryId(
+              request.authorId(), userCountry.getId());
+      if (userChats.isEmpty()) {
+        // if this was the last chat in country for user, we remove connection with Country
+        userCountryRepository.delete(userCountry);
+      }
```

Comment:
It will be better to refactor this function

Looking at the removeConnections method, it seems to handle multiple responsibilities:

    Validating if the user is part of the chat.
    Removing the user's chat connection.
    Managing the user's association with a country.

https://chatgpt.com/share/d588e0ec-5848-4e4b-bc4f-d669e70c0097